adb_client/src/usb/adb_usb_device.rs

```diff
@@ -62,6 +98,40 @@ impl ADBUSBDevice {
         Ok(s)
     }
 
+    /// autodetect connected ADB devices and establish a connection with the
+    /// first device found
+    pub fn autodetect() -> Result<Self> {
+        retry(Fixed::from_millis(3000).take(5), || {
+            let Some((vid, pid)) = search_adb_devices() else {
+                return Err(RustADBError::ADBRequestFailed(format!(
+                    "no USB devices found that match the signature of an ADB device"
+                )));
+            };
+            log::trace!("Trying to connect to ({vid}, {pid})");
+            ADBUSBDevice::new(vid, pid)
+        })
+        .map_err(|e| {
+            RustADBError::ADBRequestFailed(format!("the device took too long to respond: {e}"))
+        })
+    }
+
+    /// autodetect connected ADB devices and establish a connection with the
+    /// first device found using a custom private key path
+    pub fn autodetect_with_custom_private_key(private_key_path: PathBuf) -> Result<Self> {
```

Comment:
Couldn't we rather loop over all devices in this loop and raise if we find more than 1 device available ? 

This would be more obvious what device is really chosen

Reply:
Regarding this, we're currently using a thread to look for devices every 3 seconds. It would require us to create an extra struct which implements the `Iterator` trait to yield new values. Would you like to go that route?

Reply:
I made the necessary change, you can have a look at it :)